packages/core/src/command/feature-flags/ffs-on-dynamodb.ts

```diff
@@ -70,149 +60,148 @@ export const initialFeatureFlags: FeatureFlagDatastore = {
-      existingVersion: 0,
-    },
-    skipVersionCheck: true,
-  });
-  return updatedRecord.featureFlags;
-}
+  private constructor(private readonly region: string, private readonly tableName: string) {
+    this.ddbUtils = makeDdbClient(this.region, this.tableName);
+    FeatureFlags._instance = this;
+  }
 
-export async function getFeatureFlagsRecord({
-  region,
-  tableName,
-  skipCache,
-}: {
-  region: string;
-  tableName: string;
-  skipCache?: boolean;
-}): Promise<FeatureFlagsRecord | undefined> {
-  const now = Date.now();
-  const age = now - (featureFlagsCache?.timestamp ?? 0);
-
-  if (!skipCache && featureFlagsCache && age < cacheDuration.asMilliseconds()) {
-    return featureFlagsCache.record;
+  /**
+   * Initialize the feature flags service.
+   * It has to be called before using the service.
+   */
+  public static init(region: string, tableName: string): void {
+    if (!this._instance) {
+      new FeatureFlags(region, tableName);
+    }
   }
 
-  const ddb = makeDdbClient(region, tableName);
-  try {
-    const config = await ddb._docClient
-      .query({
-        TableName: ddb._table,
-        KeyConditionExpression: `${partitionKey} = :id`,
-        ExpressionAttributeValues: { ":id": recordId },
-        ScanIndexForward: false,
-        Limit: 1,
-      })
-      .promise();
-    const record = ddbItemToDbRecord(config.Items?.[0]);
-    if (!record) return undefined;
-
-    featureFlagsCache = { record, timestamp: now };
-    log(`Updated feature flags cache, version: ${record.version}, age was ${age} millis`);
-
-    return record;
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  } catch (error: any) {
-    if (error.code === "ResourceNotFoundException") {
-      return undefined;
+  private static get instance(): FeatureFlags {
+    if (!FeatureFlags._instance) {
+      throw new MetriportError("Feature flags not initialized");
     }
-    throw error;
+    return FeatureFlags._instance;
   }
-}
 
-export async function updateFeatureFlagsRecord({
-  region,
-  tableName,
-  newRecordData,
-  skipVersionCheck = false,
-}: {
-  region: string;
-  tableName: string;
-  newRecordData: FeatureFlagsRecordUpdate;
-  skipVersionCheck?: boolean;
-}): Promise<FeatureFlagsRecord> {
-  ffDatastoreSchema.parse(newRecordData.featureFlags);
-
-  try {
-    const existingRecord = await getFeatureFlagsRecord({ region, tableName, skipCache: true });
-
-    if (
-      !skipVersionCheck &&
-      existingRecord &&
-      existingRecord.version !== newRecordData.existingVersion
-    ) {
-      throw new BadRequestError(`FFs out of sync, reload and try again`, undefined, {
-        existingVersion: existingRecord.version,
-        updateVersion: newRecordData.existingVersion,
-      });
+  public static async getFeatureFlagsRecord({
+    skipCache = false,
+  }: {
+    skipCache?: boolean;
+  } = {}): Promise<FeatureFlagsRecord | undefined> {
+    if (skipCache) await FeatureFlags._getFeatureFlagsRecord();
```

Comment:
this won't return the value, and it doesn't warm the cache. Did you mean to add a return here?

Reply:
shit, good catch on the return 🤦 

intentionally not rebuilding the cache... it was doing that before, but (1) the param is `skipCache` not `invalidateCache` and (2) the use case for that param is to load from the source when updating the FFs... so I decided to leave as `skipCache` and do what the param says.

Cool?

Reply:
yep totally makes sense.